Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx

```diff
@@ -79,44 +79,67 @@ class TRDDPLDigitizerTask : public o2::base::BaseDPLDigitizer
     std::vector<o2::trd::Digit> digits;                         // digits which get filled
     o2::dataformats::MCTruthContainer<o2::trd::MCLabel> labels; // labels which get filled
 
+    o2::InteractionTimeRecord lastTime;    // the time of the previous event
+    o2::InteractionTimeRecord currentTime; // the current time
+    o2::InteractionTimeRecord triggerTime; // the time at which the TRD start reading out a signal
+    bool firstEvent = true;                // Flag for the first event processed
+
     TStopwatch timer;
     timer.Start();
-
     // loop over all composite collisions given from context
     // (aka loop over all the interaction records)
     for (int collID = 0; collID < irecords.size(); ++collID) {
-      mDigitizer.setEventTime(irecords[collID].getTimeNS());
+      currentTime = irecords[collID];
+      // Trigger logic implemented here
+      if (firstEvent) {
+        triggerTime = currentTime;
+        lastTime = currentTime;
+        firstEvent = false;
+      } else {
+        if (currentTime.getTimeNS() > lastTime.getTimeNS()) {
```

Comment:
the events are strictly ordered in increasing time, even interactions in the same BC.

Reply:
You are right. This part still kept the same conditional structure as when it was inside the digitizer; there, we needed to test for this to catch the embedding events. This is one reason this code should live here and not inside the digitizer; I never saw this before. I will drop the additional conditional and also the unnecessary `lastTime` variable.